### DISCUSSION TITLE
Not sure why Prisma queries are returning any/any[]

### DISCUSSION THREAD

**Discussion by jho14** [prisma/prisma#15639] [Q&A] [answered]
We've defined a model 'Subset' in our schema.prisma along with many other models. All Prisma queries work as expected in returning their respective object types except for 'Subset' which returns type any or any[] (depending on the type of query)

Prisma Client: 4.4.0 (also occurred with 3.15.2)
Prisma Engine: 3.15.0

**Incorrect typing:**
<img width="405" alt="image" src="https://user-images.githubusercontent.com/60026007/193476868-5e0014c8-3588-4d6d-9799-197620e1cf80.png">

**An example where the typing is correct:**
<img width="567" alt="image" src="https://user-images.githubusercontent.com/60026007/193476892-5b2151b1-2d25-4478-aa35-3cf743502ead.png">

**Defined schema:**
<img width="878" alt="image" src="https://user-images.githubusercontent.com/60026007/193476913-ee63ae43-b7a1-472f-b7f6-fb4b3ce66aba.png">

**Comment by austincrim** [comment, marked as answer]
Hey @jho14 👋,

This looks like an instance of [this known bug](https://github.com/prisma/prisma/issues/12469) where specific model names clash with Prisma's generated types. The only workaround I know of is to rename the model.

Feel free to give that issue a 👍 to help our engineers prioritize.

**Reply by vladi-stevanovic** [reply]
Hello @jho14 did you perhaps have a chance to try the workaround recommended by Austin? Let us know if you still have any questions!

**Reply by jho14** [reply]
@austincrim @vladi-stevanovic , the workaround did work! Adding a 👍 to the original bug report to avoid this situation in the future as Prisma expands their generated types.

**Reply by vladi-stevanovic** [reply]
That's great to hear! Thank you for letting us know!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
